src/gui/processing/qgsprocessingmeshdatasetwidget.cpp

```diff
@@ -740,4 +750,120 @@ void QgsProcessingMeshDatasetTimeWidget::updateValue()
   buildValue();
 }
 
+
+QgsProcessingMeshDatasetGroupsParameterDefinitionWidget::QgsProcessingMeshDatasetGroupsParameterDefinitionWidget(
+  QgsProcessingContext &context,
+  const QgsProcessingParameterWidgetContext &widgetContext,
+  const QgsProcessingParameterDefinition *definition,
+  const QgsProcessingAlgorithm *algorithm,
+  QWidget *parent )
+  : QgsProcessingAbstractParameterDefinitionWidget( context, widgetContext, definition, algorithm, parent )
+{
+  QVBoxLayout *vlayout = new QVBoxLayout();
+  vlayout->setContentsMargins( 0, 0, 0, 0 );
+
+  vlayout->addWidget( new QLabel( tr( "Linked input" ) ) );
+
+  mParentLayerComboBox = new QComboBox();
+  vlayout->addWidget( mParentLayerComboBox );
+
+  if ( QgsProcessingModelAlgorithm *model = widgetContext.model() )
+  {
+    const QMap<QString, QgsProcessingModelParameter> components = model->parameterComponents();
+    for ( auto it = components.constBegin(); it != components.constEnd(); ++it )
+    {
+      const QgsProcessingParameterDefinition *def = model->parameterDefinition( it.value().parameterName() );
+      if ( def && def->type() == QgsProcessingParameterMeshLayer::typeName() )
+      {
+        mParentLayerComboBox->addItem( def->description(), def->name() );
+      }
+    }
+  }
+
+  const QgsProcessingParameterMeshDatasetGroups *datasetGroupDef =
+    static_cast< const QgsProcessingParameterMeshDatasetGroups *>( definition );
+  int currentIndex = mParentLayerComboBox->findData( datasetGroupDef->meshLayerParameterName() );
+  if ( currentIndex != -1 )
+    mParentLayerComboBox->setCurrentIndex( currentIndex );
```

Comment:
elsewhere we do this:

```suggestion
    else if ( !datasetGroupDef->meshLayerParameterName().isEmpty() )
    {
        // if no layer parameter candidates found, we just add the existing one as a placeholder
        mParentLayerComboBox->addItem( datasetGroupDef->meshLayerParameterName(), datasetGroupDef->meshLayerParameterName() );
        mParentLayerComboBox->setCurrentIndex( mParentLayerComboBox->count() - 1 );
    }
```